### DISCUSSION TITLE
PyTorch is not upgrading in conda environment

### DISCUSSION THREAD

**Discussion by mhadnanali** [pyg-team/pytorch_geometric#3212] [Q&A]
I am a client user of my research lab server, There is PyTorch 1.3.1 is installed on the server but I want to use 1.8.0 with cuda 10.1. 
So when I run the command. 

"pip install torch-scatter torch-sparse torch-cluster torch-spline-conv torch-geometric -f https://data.pyg.org/whl/torch-1.8.0+cu101.html"
It says that requirements are satisfied, while when I check the version it says "Version 1.3.1" 
Why this contradiction ?

I also tried to install it with conda command, but in that way, it don't install **Cuda**, seems like my only option is pip and it also is not working. 

Take a look at the screenshot for all relevant details. 

![image](https://user-images.githubusercontent.com/15828557/134469981-f0f53331-c6d9-4065-a918-74b1734daa27.png)

**Comment by rusty1s** [comment]
The `pip install torch-scatter torch-sparse torch-cluster torch-spline-conv torch-geometric -f https://data.pyg.org/whl/torch-1.8.0+cu101.html` command does not will install PyTorch. You have to manually install it following the guidelines [here](https://pytorch.org/get-started/locally/). In case you want to force an upgrade of certain packages, it's best to use `pip install --upgrade torch-scatter ...`.

**Reply by mhadnanali** [reply]
I installed PyTorch but it doesn't seem like working. Apparently, PyTorch is installed but "import torch_geometric or import torch_geometric.transforms" is still not working. 
**Error**: OSError: libtorch_cpu.so: cannot open shared object file: No such file or directory

**Error**: OSError: /home/adnan/.local/lib/python3.8/site-packages/torch_sparse/_version_cuda.so: undefined symbol: _ZN3c106detail12infer_schema20make_function_schemaENS_8ArrayRefINS1_11ArgumentDefEEES4_

Is this a version issue or what? as my PyTorch is 1.7.1 but Pytorch geometric command had "torch-1.8.0+cu101", 
if this is a version issue then how I am suppose to find the torch 1.7.1 with cuda 10.1? 

Big Thanks for your help. 

**Note: nvcc --version and print(torch.version.cuda) version both are same.** 

![image](https://user-images.githubusercontent.com/15828557/134528331-d42027eb-2684-402b-9abf-bb2f39e795e7.png)

**Reply by rusty1s** [reply]
For installing PyG for PyTorch 1.7.1, run:
```
pip install torch-scatter==2.0.7 torch-sparse==0.6.9 torch-cluster==1.5.9 torch-spline-conv==1.2.1 torch-geometric -f https://data.pyg.org/whl/torch-1.7.1+cu101.html
```
You can find all wheels related to PyTorch 1.7.1+cu101 [here](https://data.pyg.org/whl/torch-1.7.0%2Bcu101.html)
